### DISCUSSION TITLE
Query by language

### DISCUSSION THREAD

**Discussion by chantlong** [directus/directus#14559] [Archive]
I followed the Directus official tutorial on adding translations from below, which was super helpful in adding translations.
https://www.youtube.com/watch?v=HGQppTT23T8&ab_channel=Directus

I currently only have 3 languages.
<img width="810" alt="Screen Shot 2022-07-22 at 15 39 02" src="https://user-images.githubusercontent.com/17346712/180378787-feccb8a9-fbb7-48eb-8652-04342b6c4d6b.png">
<img width="828" alt="Screen Shot 2022-07-22 at 15 45 28" src="https://user-images.githubusercontent.com/17346712/180379962-2eae602d-094c-49ad-9145-743bc06b1d58.png">
<img width="851" alt="Screen Shot 2022-07-22 at 15 45 35" src="https://user-images.githubusercontent.com/17346712/180379970-17137075-c5f7-4344-b418-a620e0caec90.png">

I am running it locally on docker and if I query by 

```
http://0.0.0.0:8055/items/test_import?fields=*.*
```

I get the result

```
{
    "data": [
        {
            "id": "6139de17-2fdd-4b34-8f57-e3382724eeb2",
            "parent": null,
            "translations": [
                {
                    "id": 4,
                    "test_import_id": "6139de17-2fdd-4b34-8f57-e3382724eeb2",
                    "languages_id": "en-US",
                    "type": "Fruit",
                    "value": "Apple"
                },
                {
                    "id": 6,
                    "test_import_id": "6139de17-2fdd-4b34-8f57-e3382724eeb2",
                    "languages_id": "de-DE",
                    "type": "Obst",
                    "value": "Apfel"
                },
                {
                    "id": 7,
                    "test_import_id": "6139de17-2fdd-4b34-8f57-e3382724eeb2",
                    "languages_id": "es-ES",
                    "type": "Fruta",
                    "value": "Manzana"
                }
            ]
        }
    ]
}
```

I looked around and let's say I want to only retrieve the `en-US` one, I tried running the query

```
http://0.0.0.0:8055/items/test_import?fields=*.*&deep[translations][_filter][languages_code][_eq]=en-US
```

But it gave me

```
{
    "errors": [
        {
            "message": "An unexpected error occurred.",
            "extensions": {
                "code": "INTERNAL_SERVER_ERROR"
            }
        }
    ]
}
```

So the question is how can I query by `en-US` only then?

**Comment by br41nslug** [comment]
Looking at the data structure of your translations it seems that the language field is called `languages_id` instead of `languages_code`. So with a bit of luck this small change will do the trick `http://127.0.0.1:8055/items/test_import?fields=*.*&deep[translations][_filter][languages_id][_eq]=en-US`

**Reply by chantlong** [reply]
Thank you!
Maybe it would be good to update the docs for Global Parameters Deep to use `language_id` since, that is the default one created when creating Translations.

https://docs.directus.io/reference/query/#deep
